### PR TITLE
Create cells for pickup game list

### PR DESCRIPTION
**Pull request by YKo20010**
## Overview
Added and structured cells for viewing various pickup games.
Added collection view to sports view controller to debug cells.
Updated Posts.swift model to match information needed for cells.

## Test Coverage
Compiles and runs. See screenshot below.

## Next Steps (delete if not applicable)
Add targets to the cells and button.

## Screenshots (delete if not applicable)
image

### CODE REVIEW

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    // MARK: - Other Views
+    private let statusButton = UIButton()
+    private let playerIcon = UIImageView()
+    
+    // MARK: - Data
+    private var status: GameStatus = .open
+    
+    // TODO: add delegate for toggling game status
+    
+    enum GameStatus {
```
Comment (CameronHamidi) [inline Uplift/Views/Cells/PickupGameCell.swift]: Can we make this a private enum?

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    }
+    
+    @objc func toggleStatus() {
+        // TODO: toggle game status
+    }
+
+    private func setupConstraints() {
+        let smallPadding = 6
+        let mediumPadding = 12
+        let largePadding = 16
```
Comment (CameronHamidi) [inline Uplift/Views/Cells/PickupGameCell.swift]: Can we alphabetize these constants?

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            make.leading.trailing.equalToSuperview().offset(mediumPadding)
+        }
+        
+        statusButton.snp.updateConstraints { make in
+            make.trailing.equalToSuperview().offset(mediumPadding)
+            make.centerY.equalToSuperview()
+            make.height.equalTo(24)
+            switch status {
+            case .created:
+                make.width.equalTo(75)
```
Comment (CameronHamidi) [inline Uplift/Views/Cells/PickupGameCell.swift]: Replace these raw ints with constants

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        }
+        
+        playersLabel.snp.updateConstraints { make in
+            make.bottom.trailing.equalToSuperview().offset(mediumPadding)
+        }
+        
+        playerIcon.snp.updateConstraints { make in
+            make.centerY.equalTo(playersLabel)
+            make.trailing.equalTo(playersLabel.snp.leading).offset(smallPadding)
+            make.height.equalTo(12)
```
Comment (CameronHamidi) [inline Uplift/Views/Cells/PickupGameCell.swift]: Same here

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+//  Uplift
+//
+//  Created by Artesia Ko on 2/19/20.
+//  Copyright © 2020 Cornell AppDev. All rights reserved.
+//
+
+import SnapKit
+import UIKit
+
+class PickupGameCell: UICollectionViewCell {
```
Comment (kevinchan159) [inline Uplift/Views/Cells/PickupGameCell.swift]: newline after line 12

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+                make.width.equalTo(75)
+            case .joined:
+                make.width.equalTo(65)
+            case .open:
+                make.width.equalTo(48)
+            }
+        }
+        
+        playersLabel.snp.updateConstraints { make in
+            make.bottom.trailing.equalToSuperview().offset(mediumPadding)
```
Comment (kevinchan159) [inline Uplift/Views/Cells/PickupGameCell.swift]: do you mean `inset` or `offset` here? `offset` would make it go outside of `contentView`

File: Uplift/Views/Cells/PickupGameCell.swift [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+import SnapKit
+import UIKit
+
+class PickupGameCell: UICollectionViewCell {
+    // MARK: - Labels
+    private let locationLabel = UILabel()
+    private let playersLabel = UILabel()
+    private let titleLabel = UILabel()
+    private let detailLabel = UILabel()
```
Comment (lucyxubroad) [inline Uplift/Views/Cells/PickupGameCell.swift]: alphabetize?

File: Uplift/Views/Cells/PickupGameCell.swift (hunk 1 above)
Comment (YKo20010) [reply Uplift/Views/Cells/PickupGameCell.swift]: We need to use it in the configure function, ie. need to access from view controller itself when we dequeue cells, so no.

Comment (YKo20010) [reply Uplift/Views/Cells/PickupGameCell.swift]: actually nvm I'm going to move the enum to models

File: Uplift/Controllers/SportsFeedViewController.swift [hunk 8]
```diff
@@ -35,6 +62,36 @@ class SportsFeedViewController: UIViewController {
             make.leading.trailing.top.equalToSuperview()
             make.height.equalTo(headerViewHeight)
         }
+        collectionView.snp.makeConstraints { make in
+            make.leading.trailing.bottom.equalToSuperview()
```
Comment (kevinchan159) [inline Uplift/Controllers/SportsFeedViewController.swift]: just double checking, do we wanna do `safeAreaLayoutGuide` for bottom?

File: Uplift/Controllers/SportsFeedViewController.swift [hunk 9]
```diff
@@ -35,6 +62,36 @@ class SportsFeedViewController: UIViewController {
...
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return posts.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: sportIdentifier, for: indexPath) as! PickupGameCell
+        let post = posts[indexPath.item]
+        let status = post.gameStatus == "JOINED" ? GameStatus.joined
+            : post.gameStatus == "CREATED" ? GameStatus.created
+            : GameStatus.open
```
Comment (kevinchan159) [inline Uplift/Controllers/SportsFeedViewController.swift]: this is kind of hard to read. can we use a switch:
```
var status: GameStatus
switch post.gameStatus {
  case "JOINED":
    status = .joined
  case "CREATED":
    status = .created
  default:
    status = .open
}
```

File: Uplift/Controllers/SportsFeedViewController.swift [hunk 10]
```diff
@@ -35,6 +62,36 @@ class SportsFeedViewController: UIViewController {
...
+        cell.configure(postId: post.id, title: post.title, type: post.type, time: post.time, location: post.location, players: post.players, gameStatus: status)
```
Comment (kevinchan159) [inline Uplift/Controllers/SportsFeedViewController.swift]: can we simplify by just passing in the post object itself?

**Pull request merged**